### ISSUE TITLE
Get dev critique of navigation

### ISSUE THREAD

**Issue by gemfarmer** [18F/18f.gsa.gov#2127]
I created [navigation.yml](https://github.com/18F/18f.gsa.gov/blob/master/_data/navigation.yml) with the hope of reducing the potential redundancy and error and simplifying the site. While it has achieved this, I am a little concerned about the complexity of the templates, and how brittle/error prone they might be as we try to future proof the site for an absence of dev support.

## This is issue is done when:
~~- [ ] Ask about this in CM's office hours~~
~~- [ ] Get feedback in jekyll channel about Jekyll site architecture~~
- [x] proper documentation about how to add page-specific navigation overrides

**Comment by coreycaitlin** [comment]
Now that we settled on long-scroll pages, I wonder if we could make the side-nav on those pages work more like [USEITI subnav](https://github.com/18F/doi-extractives-data/blob/dev/_layouts/state-page.html), where we handle the items in the frontmatter, rather than a separate file?

**Comment by gemfarmer** [comment]
@coreycaitlin I checked it out and you can add 1-level-deep subnavigation in layouts that support the subnav. You can copy the code from `_data/navigation.yml` for a piece of subnavigation like so, `nav_items` being the hook:


```yml
---
title: About 18F
path: about
permalink: /about/
layout: default-intro
lead: We help other government agencies build, buy, and share technology products.
banner_cta: true
nav_items:
-
  text: 'Our principles'
  permalink: /about/#our-principles\
  in_subnav: true
-
  text: 'Our team'
  permalink: /about/#our-team
  in_subnav: true
-
  text: 'History and funding'
  permalink: /about/#history-and-funding
  in_subnav: true
-
  text: 'For press'
  permalink: /about/#for-press
  in_subnav: true
---
```

I'll add this to a sample page and/or the docs

**Comment by gemfarmer** [comment]
Also, it is worth noting that the in-document nav_items will override general navigation specified in `navigation.yml`

**Comment by gemfarmer** [comment]
Closes with #2314
